tests/matrix.rs

```diff
@@ -97,6 +99,26 @@ pub mod matrix2 {
     }
 
     #[test]
+    fn test_sum_matrix() {
+        // This doesn't work because there isn't an impl for &Matrix2<f64>.
+        // let res: Matrix2<f64> = [A, B, C].iter().sum();
+
+        // This type resolves at least, but is ugly.
+        let res: Matrix2<f64> = [A, B, C].iter().cloned().sum();
```

Comment:
Well, I'd rather not have to do the `.cloned()` bit. I wonder if it should be implemented for the reference types as well…